### ISSUE TITLE
Setting up dev environment `gulp not found`

### ISSUE THREAD

**Issue by ilyamochalov** [Microsoft/vscode-mssql#1140]
**Unable to install npm packages**

    MSSQL Extension Version: master at d99f5bf
    VSCode Version: 1.27.1
    OS Version: Windows 10

**My system** 
```
> node -v
v8.11.1
> npm -v
5.6.0
>npm list -g --depth=0
 ...\Roaming\npm
....
+-- generator-code@1.1.36
+-- global-modules@1.0.0
+-- typescript@3.0.1
+-- vsce@1.48.0
+-- windows-build-tools@3.1.0
`-- yo@2.0.5
```

I installed gulp-cli , then do the other steps from the [wiki](https://github.com/Microsoft/vscode-mssql/wiki/contributing#gulp)
```
> npm install gulp-cli@1.2.2 -g
> npm install --greedy
> npm run install-packages    <--- errors here
+ gulp@4.0.0-alpha.3
updated 1 package in 266.163s
npm WARN rollback Rolling back minimist@1.2.0 failed (this is probably harmless): EPERM: operation not permitted, scandir 'D:\Projects\vscode-mssql\node_modules\fsevents\node_modules\rc\node_modules'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

+ gulp-install@0.6.0
removed 252 packages and updated 1 package in 139.171s
[13:47:50] Local gulp not found in D:\Projects\vscode-mssql
[13:47:50] Try running: npm install gulp
```

**Comment by ilyamochalov** [comment]
Alright I used 
```
> npm install gulp
```
Then:
```

```

**Comment by MattIrv** [comment]
I'm not able to reproduce this on my machine, though I'm using a slightly newer version of npm (6.3.0) and gulp-cli (1.3.0). If you're still having trouble, try upgrading to those versions, running `git clean -ffxd` in the vscode-mssql directory in order to clean out any old build files, and then follow the instructions again

**Comment by eashi** [comment]
I had the same problem with the same versions you have @ilyamochalov, after installing the latest npm 6.4.1, everything went fine (I had to reset my git repo and delete my node_modules folder first).

**Comment by eashi** [comment]
It stopped working for me again for a new setup. 

My environment:
```
node -v
v8.12.0
npm -v
6.4.1
```
I have noticed that the gulp tag/branch (4.0) in the package.json is not valid anymore (if you search the [gulp repo](https://github.com/gulpjs/gulp/releases) you wont find it). It will still work for machines that have already installed gulp when this tag was there.

I created a [PR 1154](https://github.com/Microsoft/vscode-mssql/pull/1154) to fix this, which passed that error mark, but got different errors due to d.ts conflicts.

I believe this issue deserves some love as it is preventing new developers from joining and help.
